code/modules/clothing/masks/gasmask.dm

```diff
@@ -2,25 +2,76 @@
 	name = "gas mask"
 	desc = "A face-covering mask that can be connected to an air supply. While good for concealing your identity, it isn't good for blocking gas flow." //More accurate
 	icon_state = "gas_alt"
-	clothing_flags = BLOCK_GAS_SMOKE_EFFECT | MASKINTERNALS
+	clothing_flags = BLOCK_GAS_SMOKE_EFFECT | MASKINTERNALS | GAS_FILTERING
 	flags_inv = HIDEEARS|HIDEEYES|HIDEFACE|HIDEFACIALHAIR|HIDESNOUT
 	w_class = WEIGHT_CLASS_NORMAL
 	inhand_icon_state = "gas_alt"
 	gas_transfer_coefficient = 0.01
 	permeability_coefficient = 0.01
 	flags_cover = MASKCOVERSEYES | MASKCOVERSMOUTH | PEPPERPROOF
 	resistance_flags = NONE
+	var/max_filters = 1
+	var/current_filters = 0
+	var/list/gas_filters
+
+/obj/item/clothing/mask/gas/examine(mob/user)
+	. = ..()
+	if(max_filters > 0)
+		. += "<span class='notice'>The amount of slots for filters are [max_filters].</span>"
+	if(current_filters > 0)
+		. += "<span class='notice'>Currently there are [current_filters] filters with [get_filter_durability()]% durability.</span>"
+
+/obj/item/clothing/mask/gas/attackby(obj/item/I, mob/user)
+	. = ..()
+	if(!istype(I, /obj/item/gas_filter))
+		return TRUE
+	if(current_filters >= max_filters)
+		return TRUE
+	if(!user.transferItemToLoc(I, src))
+		return TRUE
+	current_filters++
+	LAZYADD(gas_filters, I)
+
+/obj/item/clothing/mask/gas/consume_filter()
+	if(current_filters <= 0 || max_filters == 0)
+		return FALSE
+	var/obj/item/gas_filter/gas_filter = pick(gas_filters)
+	gas_filter.reduce_filter_status()
+	if(gas_filter.filter_status <= 0)
+		current_filters--
+		LAZYREMOVE(gas_filters, gas_filter)
+		qdel(gas_filter)
+	return TRUE
+
+/obj/item/clothing/mask/gas/proc/get_filter_durability()
+	var/max_filters_durability = current_filters * 100
+	var/current_filters_durability
+	for(var/obj/item/gas_filter/gas_filter in gas_filters)
+		current_filters_durability += gas_filter.filter_status
+	var/durability = (current_filters_durability / max_filters_durability) * 100
+	return durability
+
+/obj/item/gas_filter
+	name = "atmospheric gas filter"
+	desc = "piece of filtering cloth to be used with atmospheric gas masks and emergency gas masks"
+	icon = 'icons/obj/clothing/masks.dmi'
+	icon_state = "gas_atmos_filter"
+	var/filter_status = 100
```

Comment:
This line does not meet our contribution standards, specifically for [dmdoc compliance](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md), and should be refactored.